### ISSUE TITLE
Add reusable code/functionality for three way data binding.

### ISSUE THREAD

**Issue by rquast** [js-data/js-data#155]
It would be handy if JSData contained some framework agnostic code for dealing with three way data binding. It would be good if you could select from long poll, sse or sockets. Or should this be something created as a project separate from JSData?

Would like to see similar functionality to what pouchdb has. The advantage of doing it with something like JSData is that you can control what gets synchronized, and not synchronize the entire database if it isn't necessary.

**Comment by rquast** [comment]
I should add further that I'm using JSData on both the client and server side. I'm using rethinkdb, and the way I'd currently do this is by implementing some code that listens to the changes stream in rethinkdb (http://rethinkdb.com/api/javascript/changes/) and emit it with socket.io. On the client side you'd check for update, delete, create and trigger DS.eject, refresh, etc. Not that this is a problem having to write it all, but it would be nice to have it as part of the API, including the ability for JSData to deal with triggers (changes).

**Comment by jmdobry** [comment]
I've been thinking about ways that JSData could eliminate some of the boilerplate required to implement some of this, and I would appreciate any suggestions on implementation.

**Comment by jmdobry** [comment]
I should add that I think that solving this is more involved than simply adding a feature. It may take some re-architecting to make plugins and JSData extensions easier to create. It may very well be that there needs to be another project that uses JSData internally for this syncing functionality.

**Comment by rquast** [comment]
Yes, I think another project would make more sense. However, I think trigger support might be appropriate for JSData, so you don't need to write in trigger support for all the various adapters in the external project.

**Comment by rquast** [comment]
Or it could be that you make adapters that just do trigger support for the external framework.

**Comment by rquast** [comment]
Just thinking for a minute... a message queue that communicates between the server and the browser would be a better idea. That way you'd be sure you don't miss any trigger events.

**Comment by rquast** [comment]
Something like this, maybe: https://github.com/slyadams/socket.io-queue

**Comment by jmdobry** [comment]
For the foreseeable future, this falls outside of the scope of js-data.js

Recipes or examples may be in order, so I will keep this issue but change its scope.
